scylla-cql/src/frame/response/result.rs

```diff
@@ -769,8 +769,15 @@ pub fn deser_cql_value(typ: &ColumnType, buf: &mut &[u8]) -> StdResult<CqlValue,
                     buf.len()
                 )));
             }
-            let uuid = uuid::Uuid::from_slice(buf).expect("Deserializing Uuid failed.");
-            CqlValue::Timeuuid(uuid)
+            let timeuuid = match CqlTimeuuid::from_slice(buf) {
+                Ok(timeuuid) => timeuuid,
+                Err(e) => {
+                    return Err(ParseError::BadIncomingData(format!(
+                        "Timeuuid parsing failed: {e}"
+                    )))
+                }
+            };
+            CqlValue::Timeuuid(timeuuid)
```

Comment:
Please verify whether Scylla (and Cassandra) validates timeuuids like this, i.e. whether it allows inserting uuids that are not necessarily timeuuids into timeuuid column. If they are then perhaps we can do this validation here as well, otherwise I would abstain from it so that users don't run into a situation where they cannot read data they inserted earlier or via another driver.

Perhaps it is safer not to validate here anyway - even if the DBs properly validate right now, just in case if validation breaks in Scylla/Cassandra at some future release.

Reply:
> Please verify whether Scylla (and Cassandra) validates timeuuids like this, i.e. whether it allows inserting uuids that are not necessarily timeuuids into timeuuid column.

A simple test case shows that Scylla validates whether uuid inserted to timeuuid column is a valid timeuuid.
```rust
#[tokio::test]
async fn test_invalid_timeuuid() {
    let session: Session = create_new_session_builder().build().await.unwrap();
    let ks = unique_keyspace_name();

    session
        .query(
            format!(
                "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = \
            {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1}}",
                ks
            ),
            &[],
        )
        .await
        .unwrap();
    session.use_keyspace(ks, false).await.unwrap();

    session
        .query(
            "CREATE TABLE tab (p int, t timeuuid, PRIMARY KEY (p, t))",
            (),
        )
        .await
        .unwrap();

    // Invalid timeuuid (version 2 uuid)
    let invalid_timeuuid = uuid::uuid!("00000000-0000-2000-8080-808080808080");

    session
        .query("INSERT INTO tab (p, t) VALUES (0, ?)", (invalid_timeuuid,))
        .await
        .unwrap();
}
```

The result:
```
thread 'transport::cql_types_test::test_invalid_timeuuid' panicked at scylla/src/transport/cql_types_test.rs:1731:10:
called `Result::unwrap()` on an `Err` value: DbError(Invalid, "Exception while binding column t: marshaling error: Unsupported UUID version (2)")
```



> Perhaps it is safer not to validate here anyway - even if the DBs properly validate right now, just in case if validation breaks in Scylla/Cassandra at some future release.

But I agree, I'll get rid of the validation part in `CqlTimeuuid` constructors i.e. replace
```rust
impl TryFrom<Uuid> for CqlTimeuuid {
    type Error = InvalidTimeuuidVersion;

    fn try_from(value: Uuid) -> Result<Self, Self::Error> {
        let version = value.get_version_num();
        if version != 1 {
            return Err(InvalidTimeuuidVersion(version));
        }

        Ok(Self(value))
    }
}
```
with
```rust
impl From<Uuid> for CqlTimeuuid { ... }
```